pkg/virt-handler/vm.go

```diff
@@ -559,6 +559,10 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 			for _, existingInterfaceSpec := range vmi.Spec.Domain.Devices.Interfaces {
 				existingInterfacesSpecByName[existingInterfaceSpec.Name] = existingInterfaceSpec
 			}
+
+			interfacesNameByMac := indexSpecIfaceNamesByMac(existingInterfacesSpecByName)
+			setMissingDomainInterfacesStatusNames(interfacesNameByMac, domainInterfaceStatusByMac)
```

Comment:
About the 2nd bullet.
it will need to filter them by by type according spec, and kinda break the simpleness imo,
also it would affect only the devices which arent on the vmi.Spec.Domain.Devices.Interfaces (i.e HostDev), else they would already being handled by the code above line 625

Reply:
It was hard for me to understand you answer but I will try to reply to what I understand.
Those two lines are a workaround for *sriov only* there is no reason those methods would be called for any other interface type.
I don't find the current version simpler. I find it confusing (future developer may ask himself, why are we reading the name from the vm spec, we already have a mechanism putting the name on the domain status [1](https://github.com/kubevirt/kubevirt/pull/4335/commits/37900324ee59ff0b3056aa86afe3cea5ac2fb83f#diff-7a3f9e0461f554b07ac195375eb0f8db7d46579071795a65c459c721b1a8204aR595), [2](https://github.com/kubevirt/kubevirt/pull/4335/commits/37900324ee59ff0b3056aa86afe3cea5ac2fb83f#diff-7a3f9e0461f554b07ac195375eb0f8db7d46579071795a65c459c721b1a8204aR604), [3](https://github.com/kubevirt/kubevirt/blob/aa8a15d540ad90d7022be59b30840d41581f52bc/pkg/virt-launcher/virtwrap/agent-poller/agent_parser.go#L223)).

Reply:
Links 1 2 3 doesnt work, 
but ok will try to filter upon sriov

Reply:
Fixed link 1 and 2. 3 was working.

Reply:
For me it didnt, now it does

Reply:
About the first bullet,
i think we shouldnt move it because (line 618)
https://github.com/kubevirt/kubevirt/pull/4335/files/e5a233de0e66c0a2ad11f1232f871f87d72c294d..37900324ee59ff0b3056aa86afe3cea5ac2fb83f#diff-7a3f9e0461f554b07ac195375eb0f8db7d46579071795a65c459c721b1a8204aR618
There is logic there that removed the interfaces from domainInterfaceStatusByMac

so it think its more healthy to leave logic in the current position (orthogonal to adding the filter of sriov, which what i am doing)

Reply:
The removal you're talking about is done to mark the interface is already handled and there is a status for it. That's exactly why I ask you to move your code to be after this removal, since you code is relevant only for the un-handled (sriov) interfaces.

Reply:
I am familiar with the reason,
but i think that its better to decorate (filtered by sriov, as already pushed) the domainInterfaceStatusByMac before the removal, 
let the removal do its magic, and then the leftovers are handled on block 625.
Instead not decorate it
let the magic happens,
find just the sriovs, decorate them, and apply block 625
since the block is used already on the removal, i find it wrong to alter it in the middle.

what do you think?